Generate hw profile

m-iwanicki opened the pull request:
Tests and keywords were taken from `dasharo-compatibility/dasharo-tools-suite.robot` and minimally modified.

m-iwanicki committed:
dts: Add profile generation tests

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
docs: dts-tests: format example command as it was too long

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

DaniilKl commented:
Lets hope it will work out-of-the-box :crossed_fingers: .

m-iwanicki committed:
dts: dts-gen: move test setup to separate keyword

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki commented:
@DaniilKl obviously it doesn't and as always because of some other keywords. This time `Get Hostname Ip` can't get IP in DTS (`net-tools 2.10` hostname doesn't have `-I` argument).

m-iwanicki committed:
network: Get Hostname IP: try to get IP via 'ip a' if hostname fails

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki commented:
using `ip a` to get ip and the same regexp returns:

```sh
${ip_address} = ['192.168.4.126', '192.168.4.255', '192.168.4.223', '192.168.4.255']
```

Not sure if filtering out `.255` addresses is needed.

DaniilKl commented:
> Not sure if filtering out .255 addresses is needed.

I am not sure what the `Returns local IP address of the DUT.` of the KW means, but to me it seems like it needs only the DUT IP, not subnetwork broadcast IP.

m-iwanicki committed:
dts: dts-gen: start sshd before trying to get file from DUT

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki commented:
We take first ip from this list. I meant if I should filter those addresses before that, just in case regexp would return `[192.168.10.255, 192.168.10.5, 192.168.10.255]`.

m-iwanicki committed:
lib: network: add optional hash verification in 'Get File From DUT'

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
dts: dts-gen: don't verify files when globing

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki commented:
@DaniilKl generating profile for SeaBIOS update on APU2 works, profile is identical except it contains 1 additional dmidecode, which happens if there is one additional UI refresh. I think this part can be fixed manually or if this happens in all workflows then I can remove last line from profile in test itself:

```diff
--- "dts/profiles/pcengines-apu2 SeaBIOS Update - DPP.profile"  2025-08-13 12:23:54.184649406 +0200
+++ "dts/dts-gen-profiles/2025-08-13 17:37:40-DTG002.001 Generate Profile for DTS SeaBIOS Update Workflow/profile" 2025-08-13 17:37:44.250140860 +0200
@@ -39,2 +39,3 @@
 reboot  0
 dmidecode  0
+dmidecode  0
```

m-iwanicki commented:
To generate profile I:

- Built dts with neeeded dts-scripts changes
- Generated ipxe script:

    ```sh
    ./scripts/ci/ipxe-run.sh ~/projects/yocto/dts/ipxe/v6.6.6/{dts-base-image-v6.6.6.cpio.gz,bzImage-v6.6.6}
    ```

- Started test:

    ```
    robot -b command_log.txt -v snipeit:yes -L TRACE -v config:pcengines-apu2 \
        -v rte_ip:192.168.10.172 \
        -v dpp_email:'EMAIL' -v dpp_password:'PASSWORD' \
        -v fw_file:"$HOME/downloads/pcengines_apu2_seabios_v24.05.00.01.rom" \
        -t "DTG002*" dts/dts-gen-profile.robot
    ```

SeaBIOS and initial deployment tests need us to do some steps manually, in the case of update I had to boot into DTS.
For UEFI tests you can add `-v boot_dts_from_ipxe_shell:True -v dts_ipxe_link:http://192.168.4.126:4321/dts.ipxe` to arguments

m-iwanicki committed:
dts: dts-gen: Optional ME disable in case fw doesn't support it

'*_SUPPORT' variables apply to fw version defined in 'FW_VERSION' which
often are unreleased 'rc' versions.

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
dts-lib: iPXE boot: add interval between inputs

On some platforms writing on serial without delay between characters
resutls in truncated input e.g. on ODROID-H4

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
dts: dts-gen: Add SeaBIOS->UEFI and UEFI->SeaBIOS transitions

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
dts: dts-gen: Fix test IDs

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
dts: dts-gen: add missing DTS booting

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
dts: dts-gen: remove redundant 'Network Boot' check

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
dts: dts-gen: fix version transition

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
dts: dts-gen: trim last dmidecode

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki commented:
@DaniilKl I tested:

- SeaBIOS Update (APU2) - results in additional dmidecode
- SeaBIOS->UEFI transition (APU2) - no additional dmidecode
- UEFI Initial deployment (odroid) - no additional dmidecode
- UEFI update (MSI Z790) - no additional dmidecode

I added trimming last line for SeaBIOS update

m-iwanicki committed:
dts: gen-profile: add platform to folder name & trim debug_profile

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
dts: move duplicate keywords to dts-lib

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
.gitignore: add UEFI

I'm not sure why and when it's generated

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki committed:
dts-e2e: gen-profile: expand test documentation, mention fw_file variable

Signed-off-by: Michał Iwanicki <michal.iwanicki@3mdeb.com>

m-iwanicki commented:
When running those tests on platform with `DUT_CONNECTION_METHOD` == "pikvm" (e.g. MSI) make sure to set `DEVICE_IP` which is DUT IP (as seen in DTS)
